### ISSUE TITLE
Can I debug and load configMap locally?

### ISSUE THREAD

**Issue by 344089386** [tintoy/dotnet-kube-client#124]
I have loaded configMap but configuration["Redis:Configuration"] is null.I heard that configMap should be mounted to POD or Deployment. Can't you debug it locally?

![image](https://user-images.githubusercontent.com/22017199/88002387-ffe29a00-cb34-11ea-94b3-6d83ddf0e7da.png)
![image](https://user-images.githubusercontent.com/22017199/88002483-37514680-cb35-11ea-9f7f-40774aca332f.png)
![image](https://user-images.githubusercontent.com/22017199/88002454-27396700-cb35-11ea-9beb-36d034900706.png)
![image](https://user-images.githubusercontent.com/22017199/88002518-49cb8000-cb35-11ea-8500-a48a22a239ed.png)

**Comment by tintoy** [comment]
Can you check the section of the readme about how to turn on logging for the client? It should be able to log the requests it’s making which might give us a clue as to what’s going wrong...

**Comment by 344089386** [comment]
> 您可以检查自述文件中有关如何为客户端打开日志记录的部分吗？它应该能够记录正在发出的请求，这可能会为我们提供了有关发生问题的线索...

I don't know where exactly you mean. Could you be more specific?

**Comment by 344089386** [comment]
> Can you check the section of the readme about how to turn on logging for the client? It should be able to log the requests it’s making which might give us a clue as to what’s going wrong...

I don't know where exactly you mean. Could you be more specific?

**Comment by 344089386** [comment]
@tintoy      
Is this?
![image](https://user-images.githubusercontent.com/22017199/88007276-3540b500-cb40-11ea-89ad-df7976e8c439.png)
![image](https://user-images.githubusercontent.com/22017199/88007258-2823c600-cb40-11ea-8305-9864bc55163b.png)

**Comment by tintoy** [comment]
Yep - from memory you can even send the logs to Serilog as well.

**Comment by tintoy** [comment]
As long as it logs to the console or the VS debugger console it should show you the requests it’s making (there is a property on the client options to log requests.

**Comment by tintoy** [comment]
(As you guessed, it’s the logger factory property). Sorry am viewing this on my phone as I’m not near a computer right now.

Take a look at the ConfigMap sample in the repository, it logs requests and does most of what you’d need to hook up configuration from a ConfigMap.

**Comment by 344089386** [comment]
I looked at the ConfigMap sample and it seems that it still can't solve my problem.
@tintoy

**Comment by 344089386** [comment]
Do you use ConfigMap as the configuration center code?Can you send it to me for reference?

**Comment by tintoy** [comment]
Are you able to run that sample successfully?

**Comment by 344089386** [comment]
Can the.   @tintoy 
![image](https://user-images.githubusercontent.com/22017199/88146635-b2e2ee80-cc2e-11ea-8f4e-599f69943b20.png)
![image](https://user-images.githubusercontent.com/22017199/88146652-b7a7a280-cc2e-11ea-91df-114459cbc62c.png)

**Comment by tintoy** [comment]
What happens when you press enter?

**Comment by 344089386** [comment]
@tintoy 
![image](https://user-images.githubusercontent.com/22017199/88153595-335a1d00-cc38-11ea-9a01-c0d796ce5889.png)
Delete seems to delete unsuccessful, after the creation of an error.

After I manually delete, perform as follows.
![image](https://user-images.githubusercontent.com/22017199/88153928-a9f71a80-cc38-11ea-8932-6235e7c85160.png)

**Comment by tintoy** [comment]
In your own program, do you see *any* keys in the IConfiguration once it has been built? What is the name of the namespace that your config map is in?

**Comment by 344089386** [comment]
@tintoy 
Loading the local appSettings. json will work fine.  The namespace name is MicroService.

**Comment by tintoy** [comment]
It looks like in your original example code you are not specifying the namespace where the configmap lives - can you try passing the namespace in your original program and see if that works?

**Comment by 344089386** [comment]
@tintoy 
No.
![image](https://user-images.githubusercontent.com/22017199/88255694-aec8d680-cceb-11ea-94e6-227e6bfd424c.png)

**Comment by tintoy** [comment]
What keys *does* the configuration have?

**Comment by 344089386** [comment]
@tintoy 
`{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "http://localhost:65115/"
    }
  },
  "AuthServer": {
    "Authority": "http://localhost:64999",
    "ApiName": "BackendAdminAppGateway"
  },
  "ElasticSearch": {
    "Url": "http://139.224.255.200:3005"
  },
  "ConnectionStrings": {
    "Default": "HOST=60.22.155.248;PORT=30000;DATABASE=MsDemo_Identity;PASSWORD=123456;USER ID=postgres;"
  },
  "Redis": {
    "Configuration": "139.224.255.200:3002"
  },
  "ReRoutes": [
    {
      "DownstreamPathTemplate": "/api/identity/{everything}",
      "DownstreamScheme": "http",
      "UpstreamPathTemplate": "/api/identity/{everything}",
      "ServiceName": "identity-service",
      "UpstreamHttpMethod": [ "Put", "Delete", "Get", "Post" ]
    },
    {
      "DownstreamPathTemplate": "/api/multi-tenancy/{everything}",
      "DownstreamScheme": "http",
      "UpstreamPathTemplate": "/api/multi-tenancy/{everything}",
      "ServiceName": "multi-tenancy-service",
      "UpstreamHttpMethod": [ "Put", "Delete", "Get", "Post" ]
    },
    {
      "DownstreamPathTemplate": "/api/productManagement/{everything}",
      "DownstreamScheme": "http",
      "UpstreamPathTemplate": "/api/productManagement/{everything}",
      "ServiceName": "product-management-service",
      "UpstreamHttpMethod": [ "Put", "Delete", "Get", "Post" ]
    }
  ],
  "GlobalConfiguration": {
    "ServiceDiscoveryProvider": {
      "Namespace": "microservice",
      "Type": "kube"
    }
  },
  "Logging": {
    "LogLevel": {
      "Default": "Warning"
    }
  },
  "AllowedHosts": "*"
}
`

**Comment by tintoy** [comment]
Sorry I mean what keys are in the `IConfiguration`? It’s enumerable I think so you can use foreach to see all the keys.

**Comment by 344089386** [comment]
@tintoy 
Just this one key
![image](https://user-images.githubusercontent.com/22017199/88258123-f6526100-ccf1-11ea-849b-5b8c7811adac.png)

**Comment by 344089386** [comment]
@tintoy 
I'm not using it now, I'm just going to mount it, and that's fine.
Now I would like to ask how this method should be used and what needs to be configured?Is there any documentation for that

![image](https://user-images.githubusercontent.com/22017199/88516470-e69f8900-d01f-11ea-97c6-cab453ca36cb.png)

**Comment by tintoy** [comment]
Hi - there are some examples here:

https://github.com/tintoy/dotnet-kube-client#make-the-client-available-for-dependency-injection

😄

**Comment by 344089386** [comment]
@tintoy 
I don't know you used Ocelot.Provider.Kubernetes this component.
![image](https://user-images.githubusercontent.com/22017199/88518644-6418c880-d023-11ea-8596-af7bdbe988c3.png)
![image](https://user-images.githubusercontent.com/22017199/88518664-6ed35d80-d023-11ea-9472-fc16feba9333.png)

I can't find the service. When I access the gateway, an error will be reported:
![image](https://user-images.githubusercontent.com/22017199/88518801-a6daa080-d023-11ea-967c-ea21b39365d9.png)

**Comment by tintoy** [comment]
The extension method from this library is called `AddKubeClient()`, I’m not sure what `AddKubernetes()` does?

**Comment by tintoy** [comment]
I’m not sure how Ocelot’s Kubernetes support works but I’m not aware of an integration between it and this library...

**Comment by 344089386** [comment]
I decided to start ons}, as a service discovery and configuration centerNo more tossing and turning.
